### PR TITLE
Fix issue with empty folder raising error, and fix duplicate emails/ XSUP-32511

### PR DESCRIPTION
**Pull request by RosenbergYehuda**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/XSUP-32511

## Description

- Fixing an issue where fetching empty folders raises an error. 
- Hopefully, this also resolves the problem of emails being fetched twice in some cases.
- fixed an issue that default values are not in use when demisto.params are called.
- Changed the _limit_ from having a max of 200, to be the actual value the user has entered.

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/MailListener/ReleaseNotes/1_0_52.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+
+#### Integrations
+
+##### Mail Listener v2
+- Updated the Docker image to: *demisto/py3-tools:1.0.0.87415*.
+
+- Fixed an issue where fetching emails from an empty folder returned an error.
```
Comment (JasBeilin) [inline Packs/MailListener/ReleaseNotes/1_0_52.md:7]:
```suggestion
- Fixed an issue where fetching emails from an empty folder returned an error.
- Fixed an issue where we first email in a folder was skipped.
```

File: Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py [hunk 2]
```diff
@@ -623,7 +625,8 @@ def main():     # pragma: no cover
                                                       delete_processed=delete_processed, limit=limit,
                                                       save_file=save_file)
                 demisto.debug(f"{next_run=}")
-                demisto.setLastRun(next_run)
+                # if next_run is None, we will not update last_run
+                demisto.setLastRun(next_run) if next_run else None
```
Comment (JasBeilin) [inline Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py:629]:
```suggestion
                if next_run:
                    demisto.setLastRun(next_run)
```

Not sure if this is automatic fix but its confusing and not worth the one-liner.

**Review by JasBeilin — CHANGES_REQUESTED** [review]

File: Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py [hunk 3]
```diff
@@ -574,16 +575,17 @@ def main():     # pragma: no cover
     folder = params.get('folder')
     username = demisto.params().get('credentials').get('identifier')
     password = demisto.params().get('credentials').get('password')
-    verify_ssl = not params.get('insecure', False)
-    tls_connection = params.get('TLS_connection', True)
-    include_raw_body = demisto.params().get('Include_raw_body', False)
+    verify_ssl = not params.get('insecure') or True
+    tls_connection = params.get('TLS_connection') or True
```
Comment (JasBeilin) [inline Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py]: Lets revert as discussed.
```suggestion
    verify_ssl = not params.get('insecure', False)
    tls_connection = params.get('TLS_connection', True)
```

File: Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py (hunk 2 above)
Comment (RosenbergYehuda) [reply Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py:629]: Done

File: Packs/MailListener/ReleaseNotes/1_0_52.md (hunk 1 above)
Comment (RosenbergYehuda) [reply Packs/MailListener/ReleaseNotes/1_0_52.md:7]: Not exactly, the firs email issue was alredy merged in previos PR, but now that the custemur confirmed it solves the duplicate issue, I will add that.

File: Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py (hunk 3 above)
Comment (RosenbergYehuda) [reply Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py]: was reverted and discussed

File: Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py [hunk 4]
```diff
@@ -574,16 +574,16 @@ def main():     # pragma: no cover
...
+    include_raw_body = demisto.params().get('Include_raw_body') or False
     permitted_from_addresses = demisto.params().get('permittedFromAdd', '')
     permitted_from_domains = demisto.params().get('permittedFromDomain', '')
     with_headers = params.get('with_headers')
-    delete_processed = demisto.params().get("delete_processed", False)
-    limit = min(int(demisto.params().get('limit', '50')), 200)
-    save_file = params.get('save_file', False)
-    first_fetch_time = demisto.params().get('first_fetch', '3 days').strip()
+    delete_processed = demisto.params().get("delete_processed") or False
+    limit = min(int(demisto.params().get('limit') or 50), 200)
```
[the remark this answers is not in the document]
Comment (RosenbergYehuda) [reply Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py]: After our change to allow more than 200, I think we dont need this.

**Review by JasBeilin — APPROVED** [review]
Nice!

**Comment by ShirleyDenkberg** [comment]
@JasBeilin Doc review completed.

**Pull request merged**